### PR TITLE
fix bugs in split converter

### LINKED ISSUE #401: 🐛 [Bug] Split converter bugs
##  Bug Description

1. When the split operation is followed by another operation, such as the add operation, if you call the `ITensorOrFreeze ` function in add Converter, you will get an error like that:
image

2. A tensor x = at::randint(1, 10, {1, 6, 4, 4}, {at::kCUDA}),  the result's shape of aten::split(x, 2, 1) should be [{1, 2, 4, 4}, {1, 2, 4 , 4}, {1, 2, 4 , 4}], but got [{1, 1, 4, 4}, {1, 1, 4, 4}, {1, 1, 4, 4}]. It caused an error in test unit like that:
image

3. A tensor x = at::randint(1, 10, {1, 5, 4, 4}, {at::kCUDA}), the result's shape of aten::split(x, 2, 1) should be [{1, 2, 4, 4}, {1, 2, 4 , 4}, {1, 1, 4 , 4}], but got an error int test unit like that:
image

## To Reproduce

Add three test units correspondingly
```
TEST(Converters, ATenSplitAndAddConvertsCorrectly) {
  const auto graph = R"IR(
    graph(%argument_1.1 : Tensor):
          %2 : int = prim::Constant[value=2]()
          %2.1 : int = prim::Constant[value=1]()
          %3 : Tensor[] = aten::split(%argument_1.1, %2, %2.1)
          %4 : Tensor, %5 : Tensor = prim::ListUnpack(%3)
          %6 : Tensor = aten::add(%4, %5, %2.1)
          return (%6))IR";

  auto g = std::make_shared<torch::jit::Graph>();

  torch::jit::parseIR(graph, &*g);

  auto in = at::randint(1, 10, {1, 4, 4, 4}, {at::kCUDA});

  auto jit_in = at::clone(in);
  auto params = trtorch::core::conversion::get_named_params(g->inputs(), {});
  auto jit_results = trtorch::tests::util::RunGraph(g, params, {jit_in});

  auto trt_in = at::clone(in);
  auto trt_results = trtorch::tests::util::RunGraphEngine(g, params, {trt_in});

  for (size_t i = 0; i < jit_results.size(); i++) {
    auto trt = trt_results[i].reshape(jit_results[i].sizes());
    ASSERT_TRUE(trtorch::tests::util::almostEqual(jit_results[i], trt, 2e-6));
  }
}

TEST(Converters, ATenSplitFixedSplitSizeIsTwoConvertsCorrectly) {
  const auto graph = R"IR(
    graph(%argument_1.1 : Tensor):
          %2 : int = prim::Constant[value=2]()
          %2.1 : int = prim::Constant[value=1]()
          %3 : Tensor[] = aten::split(%argument_1.1, %2, %2.1)
          %4 : Tensor, %5 : Tensor, %6 : Tensor = prim::ListUnpack(%3)
          return (%4, %5, %6))IR";

  auto g = std::make_shared<torch::jit::Graph>();

  torch::jit::parseIR(graph, &*g);

  auto in = at::randint(1, 10, {1, 6, 4, 4}, {at::kCUDA});

  auto jit_in = at::clone(in);
  auto params = trtorch::core::conversion::get_named_params(g->inputs(), {});
  auto jit_results = trtorch::tests::util::RunGraph(g, params, {jit_in});

  auto trt_in = at::clone(in);
  auto trt_results = trtorch::tests::util::RunGraphEngine(g, params, {trt_in});

  for (size_t i = 0; i < jit_results.size(); i++) {
    auto trt = trt_results[i].reshape(jit_results[i].sizes());
    ASSERT_TRUE(trtorch::tests::util::almostEqual(jit_results[i], trt, 2e-6));
  }
}

TEST(Converters, ATenSplitFixedHasRemainderConvertsCorrectly) {
  const auto graph = R"IR(
    graph(%argument_1.1 : Tensor):
          %2 : int = prim::Constant[value=2]()
          %2.1 : int = prim::Constant[value=1]()
          %3 : Tensor[] = aten::split(%argument_1.1, %2, %2.1)
          %4 : Tensor, %5 : Tensor, %6 : Tensor = prim::ListUnpack(%3)
          return (%4, %5, %6))IR";

  auto g = std::make_shared<torch::jit::Graph>();

  torch::jit::parseIR(graph, &*g);

  auto in = at::randint(1, 10, {1, 5, 4, 4}, {at::kCUDA});

  auto jit_in = at::clone(in);
  auto params = trtorch::core::conversion::get_named_params(g->inputs(), {});
  auto jit_results = trtorch::tests::util::RunGraph(g, params, {jit_in});

  auto trt_in = at::clone(in);
  auto trt_results = trtorch::tests::util::RunGraphEngine(g, params, {trt_in});

  for (size_t i = 0; i < jit_results.size(); i++) {
    auto trt = trt_results[i].reshape(jit_results[i].sizes());
    ASSERT_TRUE(trtorch::tests::util::almostEqual(jit_results[i], trt, 2e-6));
  }
}
```

## Environment

> Build information about the TRTorch compiler can be found by turning on debug messages

 - PyTorch Version (e.g., 1.0):
 - CPU Architecture:
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source):
 - Build command you used (if compiling from source):
 - Are you using local sources or building from archives:
 - Python version:
 - CUDA version:
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by guoruoqian**
Signed-off-by: Ruoqian Guo <ruoqiang@nvidia.com>

# Description

Fix Split Converter bugs.  

1. Fix bug in ```core/conversion/converters/select.cpp: add_split``` function. 
2. Modified the `Var::ITensorOrFreeze` function in `core/conversion/var/Var.cpp`.

In addition, i use the linters to ensure that my code matches the style guidelines. The linters changed the entire file(core/conversion/converters/impl/select.cpp) . I only modified the `add_split ` part in select.cpp.

Fixes #401 

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/converters/impl/select.cpp [hunk 1]
```diff
@@ -27,10 +27,13 @@ bool add_split(ConversionCtx* ctx, const torch::jit::Node* n, args& args, bool s
...
+    for (int i = 0; i < numOutputs; i++) {
       sizes.push_back(split_size);
-    } else {
-      sizes = std::vector<int64_t>(numOutputs, 1);
+    }
+    if (numRemainder) {
+      numOutputs += 1;
+      sizes.push_back(numRemainder);
     }
   }
```
Comment (guoruoqian) [inline core/conversion/converters/impl/select.cpp:38]: I only modified the `add_split `function in select.cpp.

**Comment by narendasan** [comment]
Is this really a breaking change?

File: core/conversion/converters/impl/select.cpp [hunk 2]
```diff
@@ -27,10 +27,13 @@ bool add_split(ConversionCtx* ctx, const torch::jit::Node* n, args& args, bool s
   } else {
     auto split_size = args[1].unwrapToInt();
     numOutputs = inDimSize / split_size;
-    if (numOutputs == 1) {
+    numRemainder = inDimSize % split_size;
+    for (int i = 0; i < numOutputs; i++) {
```
Comment (narendasan) [inline core/conversion/converters/impl/select.cpp:31]: style thing: use int64_t if possible

File: core/conversion/var/Var.cpp [hunk 3]
```diff
@@ -90,24 +90,37 @@ nvinfer1::ITensor* Var::ITensorOrFreeze(ConversionCtx* ctx) {
...
-    out = const_layer->getOutput(0);
-
-    std::ostringstream tensor_id;
-    tensor_id << reinterpret_cast<int*>(out);
-
-    LOG_DEBUG(ctx->logger, "Freezing tensor " << tensor_id.str() << " as an IConstantLayer");
-    const_layer->setName(("[Freeze Tensor " + tensor_id.str() + " ]").c_str());
+    if (ptr_.ivalue->isTensor()) {
+      auto tensor = ptr_.ivalue->toTensor();
+      if (tensor.scalar_type() == at::kLong) {
```
Comment (narendasan) [inline core/conversion/var/Var.cpp:101]: Can we remove this? #407 is adding a formal API to support this.

File: core/conversion/var/Var.cpp [hunk 4]
```diff
@@ -90,24 +90,37 @@ nvinfer1::ITensor* Var::ITensorOrFreeze(ConversionCtx* ctx) {
...
+
+      std::ostringstream tensor_id;
+      tensor_id << reinterpret_cast<int*>(out);
+
+      LOG_DEBUG(ctx->logger, "Freezing tensor " << tensor_id.str() << " as an IConstantLayer");
+      const_layer->setName(("[Freeze Tensor " + tensor_id.str() + " ]").c_str());
+    } else {
+      // Split converter generates c10::IValue which hold TensorContainer.
+      auto output_container = ptr_.ivalue->toCustomClass<TensorContainer>();
+      out = output_container.get()->tensor();
```
Comment (peri044) [inline core/conversion/var/Var.cpp:122]: This looks good to me. 
Previously for split, we only considered cases with split as the final output or the list unpack as the final output, which is why the bug occurred for ops that followed them.

**Review by narendasan — COMMENTED** [review]
lgtm

**Pull request merged**